cnf-certification-test/lifecycle/suite.go

```diff
@@ -457,14 +498,20 @@ func testHighAvailability(env *provider.TestEnvironment) {
 
 		if dp.Spec.Template.Spec.Affinity == nil ||
 			dp.Spec.Template.Spec.Affinity.PodAntiAffinity == nil {
-			badDeployments = append(badDeployments, dp.ToString())
 			tnf.ClaimFilePrintf("Deployment found without valid high availability: %s", dp.ToString())
+			nonCompliantObjects = append(nonCompliantObjects, testhelper.NewDeploymentReportObject(dp.Namespace, dp.Name, "Deployment found without valid high availability", false))
+			compliantDeployment = false
+		}
```

Comment:
An `else` branch here might avoid the use of the `compliantDeployment` var.

Reply:
Done.